clang 17 is now required

Hello!

I tried to build Bitcoin Core again and ran into the following issue when running `cmake --build`:

```
In file included from /home/ekzyis/prog/bitcoin/src/wallet/init.cpp:22:
In file included from /home/ekzyis/prog/bitcoin/src/wallet/wallet.h:34:
/home/ekzyis/prog/bitcoin/src/wallet/transaction.h:102:23: error: no viable constructor or deduction guide for deduction of template arguments of 'Overloaded'
    return std::visit(util::Overloaded{
                      ^
/home/ekzyis/prog/bitcoin/src/util/overloaded.h:16:30: note: candidate function template not viable: requires 1 argument, but 5 were provided
template<class... Ts> struct Overloaded : Ts... { using Ts::operator()...; };
                             ^
/home/ekzyis/prog/bitcoin/src/util/overloaded.h:16:30: note: candidate function template not viable: requires 0 arguments, but 5 were provided
In file included from /home/ekzyis/prog/bitcoin/src/wallet/init.cpp:22:
In file included from /home/ekzyis/prog/bitcoin/src/wallet/wallet.h:34:
/home/ekzyis/prog/bitcoin/src/wallet/transaction.h:114:23: error: no viable constructor or deduction guide for deduction of template arguments of 'Overloaded'
    return std::visit(util::Overloaded{
                      ^
/home/ekzyis/prog/bitcoin/src/util/overloaded.h:16:30: note: candidate function template not viable: requires 1 argument, but 5 were provided
template<class... Ts> struct Overloaded : Ts... { using Ts::operator()...; };
                             ^
/home/ekzyis/prog/bitcoin/src/util/overloaded.h:16:30: note: candidate function template not viable: requires 0 arguments, but 5 were provided
2 errors generated.
make[2]: *** [src/CMakeFiles/bitcoin_node.dir/build.make:1364: src/CMakeFiles/bitcoin_node.dir/wallet/init.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:750: src/CMakeFiles/bitcoin_node.dir/all] Error 2
make: *** [Makefile:146: all] Error 2
```

Using `git bisect`, I have found out that clang 16 is no longer supported since [`faed118f`](https://github.com/bitcoin/bitcoin/commit/faed118fb30fbc303e9d4c70569abfee397f1759#diff-5724a8c914937c7dd49cc6dba4b2a62d93d5453d85867102452334ace6df8ee6).

Not sure how I was able to build [`e6851720`](https://github.com/bitcoin/bitcoin/commit/e68517208b4cc02fa4e9e6a8de0fc43b536a3b02) before in #4. It's a descendant of [`faed118f`](https://github.com/bitcoin/bitcoin/commit/faed118fb30fbc303e9d4c70569abfee397f1759#diff-5724a8c914937c7dd49cc6dba4b2a62d93d5453d85867102452334ace6df8ee6) 🤔

I'll try to fix this and will send a PR if successful!

Yeah, I'm getting the same error now when building [`e6851720`](https://github.com/bitcoin/bitcoin/commit/e68517208b4cc02fa4e9e6a8de0fc43b536a3b02):

```
In file included from /home/ekzyis/prog/bitcoin/src/wallet/init.cpp:22:
In file included from /home/ekzyis/prog/bitcoin/src/wallet/wallet.h:35:
/home/ekzyis/prog/bitcoin/src/wallet/transaction.h:102:23: error: no viable constructor or deduction guide for deduction of template arguments of 'Overloaded'
    return std::visit(util::Overloaded{
                      ^
/home/ekzyis/prog/bitcoin/src/util/overloaded.h:16:30: note: candidate function template not viable: requires 1 argument, but 5 were provided
template<class... Ts> struct Overloaded : Ts... { using Ts::operator()...; };
                             ^
/home/ekzyis/prog/bitcoin/src/util/overloaded.h:16:30: note: candidate function template not viable: requires 0 arguments, but 5 were provided
In file included from /home/ekzyis/prog/bitcoin/src/wallet/init.cpp:22:
In file included from /home/ekzyis/prog/bitcoin/src/wallet/wallet.h:35:
/home/ekzyis/prog/bitcoin/src/wallet/transaction.h:114:23: error: no viable constructor or deduction guide for deduction of template arguments of 'Overloaded'
    return std::visit(util::Overloaded{
                      ^
/home/ekzyis/prog/bitcoin/src/util/overloaded.h:16:30: note: candidate function template not viable: requires 1 argument, but 5 were provided
template<class... Ts> struct Overloaded : Ts... { using Ts::operator()...; };
                             ^
/home/ekzyis/prog/bitcoin/src/util/overloaded.h:16:30: note: candidate function template not viable: requires 0 arguments, but 5 were provided
2 errors generated.
make[2]: *** [src/CMakeFiles/bitcoin_node.dir/build.make:1364: src/CMakeFiles/bitcoin_node.dir/wallet/init.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:700: src/CMakeFiles/bitcoin_node.dir/all] Error 2
make: *** [Makefile:146: all] Error 2
```

Weird

--- a/default.nix
+++ b/default.nix
@@ -1,6 +1,6 @@
 { nixpkgs ? import builtins.fetchTarball {
-    url = "https://github.com/NixOS/nixpkgs/archive/057f9aecfb71c4437d2b27d3323df7f93c010b7e.tar.gz";
-    sha256 = "1ndiv385w1qyb3b18vw13991fzb9wg4cl21wglk89grsfsnra41k";
+    url = "https://github.com/NixOS/nixpkgs/archive/871b9fd269ff6246794583ce4ee1031e1da71895.tar.gz";
+    sha256 = "1zn1lsafn62sz6azx6j735fh4vwwghj8cc9x91g5sx2nrg23ap9k";
   } {}
 , withGui ? false
 , withWallet ? true
@@ -9,7 +9,7 @@
 }:
 {
   bitcoin = with nixpkgs; callPackage ./bitcoin.nix {
-    stdenv = clang16Stdenv;
+    stdenv = clang19Stdenv;
     inherit (qt5) qtbase qttools wrapQtAppsHook;
     inherit (darwin) autoSignDarwinBinariesHook;
     python3 = python3.withPackages (p: with p; [

--- a/shell.nix
+++ b/shell.nix
@@ -1,6 +1,6 @@
 { nixpkgs ? import (builtins.fetchTarball {
-    url = "https://github.com/NixOS/nixpkgs/archive/057f9aecfb71c4437d2b27d3323df7f93c010b7e.tar.gz";
-    sha256 = "1ndiv385w1qyb3b18vw13991fzb9wg4cl21wglk89grsfsnra41k";
+    url = "https://github.com/NixOS/nixpkgs/archive/871b9fd269ff6246794583ce4ee1031e1da71895.tar.gz";
+    sha256 = "1zn1lsafn62sz6azx6j735fh4vwwghj8cc9x91g5sx2nrg23ap9k";
   }) {}
 , withGui ? false
 , withWallet ? true
@@ -13,7 +13,7 @@ let
   };
 in
   nixpkgs.mkShell.override {
-    stdenv = nixpkgs.clang16Stdenv; # required for fuzzing; must be recent version
+    stdenv = nixpkgs.clang19Stdenv; # required for fuzzing; must be recent version
   } {
     inherit (main.bitcoin) configureFlags;
     inputsFrom = [

